service-base/src/main/java/org/eclipse/hono/service/monitoring/AbstractMessageSenderConnectionEventProducer.java

```diff
@@ -82,23 +82,29 @@ protected AbstractMessageSenderConnectionEventProducer(
             return Future.succeededFuture();
         }
 
-        return getOrCreateSender(context.getMessageSenderClient(), authenticatedDevice.getTenantId())
-                .compose(sender -> {
-
-                    final JsonObject payload = new JsonObject();
-                    payload.put("cause", cause);
-                    payload.put("remote-id", remoteId);
-                    payload.put("source", protocolAdapter);
-
-                    if (data != null) {
-                        payload.put("data", data);
-                    }
-
-                    return sender.send(
-                            authenticatedDevice.getDeviceId(),
-                            payload.encode().getBytes(StandardCharsets.UTF_8),
-                            EventConstants.EVENT_CONNECTION_NOTIFICATION_CONTENT_TYPE
-                            );
+        return context.getTenantClient().getOrCreateTenantClient()
+                .map(tenantClient -> tenantClient.get(authenticatedDevice.getTenantId()))
+                .map(tenantObject -> {
+                    return getOrCreateSender(context.getMessageSenderClient(), authenticatedDevice.getTenantId())
+                            .compose(sender -> {
+
+                                final JsonObject payload = new JsonObject();
+                                payload.put("cause", cause);
+                                payload.put("remote-id", remoteId);
+                                payload.put("source", protocolAdapter);
+
+                                if (data != null) {
+                                    payload.put("data", data);
+                                }
+
+                                final Long maxTtl = tenantObject.result().getResourceLimits().getMaxTtl();
+                                return sender.send(
```

Comment:
we can use `MessageHelper.newMessage()` here to create the message and send it using `DownstreamSender.send(Message)` ...

Reply:
`MessageHelper.newMessage()` sets the message's `orig_adapter` application property to the name of the adapter forwarding the connection event. however, the name of the adapter is already included in the message's payload. i added a `todo` comment in the source code to get some feedback

Reply:
@sophokles73 if you have time, can you take a look at rev. 4913146 ?